src/eos-metrics-instrumentation.c

```diff
@@ -413,27 +413,48 @@ record_logout_for_all_remaining_sessions (void)
 static GDBusProxy *
 login_dbus_proxy_new (void)
 {
-    GError *error = NULL;
-    GDBusProxy *dbus_proxy =
-      g_dbus_proxy_new_for_bus_sync (G_BUS_TYPE_SYSTEM,
-                                     G_DBUS_PROXY_FLAGS_NONE,
-                                     NULL /* GDBusInterfaceInfo */,
-                                     "org.freedesktop.login1",
-                                     "/org/freedesktop/login1",
-                                     "org.freedesktop.login1.Manager",
-                                     NULL /* GCancellable */,
-                                     &error);
-    if (dbus_proxy == NULL)
-      {
-        g_warning ("Error creating GDBusProxy: %s\n", error->message);
-        g_error_free (error);
-      }
-    else
-      {
-        g_signal_connect (dbus_proxy, "g-signal", G_CALLBACK (record_login),
-                          NULL /* data */);
-      }
-    return dbus_proxy;
+  GError *error = NULL;
+  GDBusProxy *dbus_proxy =
+    g_dbus_proxy_new_for_bus_sync (G_BUS_TYPE_SYSTEM,
+                                   G_DBUS_PROXY_FLAGS_NONE,
+                                   NULL /* GDBusInterfaceInfo */,
+                                   "org.freedesktop.login1",
+                                   "/org/freedesktop/login1",
+                                   "org.freedesktop.login1.Manager",
+                                   NULL /* GCancellable */,
+                                   &error);
+  if (error)
+    {
+      g_warning ("Error creating GDBusProxy: %s", error->message);
```

Comment:
This (like most of my comments) is for consistency with the rest of the metrics system.